src/main.rs

```diff
@@ -540,3 +547,180 @@ mod provision {
         Ok(())
     }
 }
+
+mod ok {
+    use super::*;
+    use colorful::*;
+    use std::arch::x86_64;
+    use std::mem::transmute;
+    use std::str::from_utf8;
+
+    #[derive(StructOpt)]
+    pub enum SevGeneration {
+        #[structopt(about = "AMD SEV Extensions")]
+        Sev,
+
+        #[structopt(about = "AMD SEV-ES Extensions")]
+        Es,
+    }
+
+    struct CpuId {
+        pub name: &'static str,
+        pub leaf: u32,
+        pub func: fn(x86_64::CpuidResult) -> (bool, Option<String>),
+    }
+
+    struct OkTestResult {
+        pub success: bool,
+        pub msg: String,
+    }
+
+    impl CpuId {
+        pub fn execute(&self) -> OkTestResult {
+            let res = (self.func)(unsafe { x86_64::__cpuid(self.leaf) });
+            let success = res.0;
+            let msg = if success {
+                if let Some(m) = res.1 {
+                    m
+                } else {
+                    format!("{}", "Supported".green())
+                }
+            } else {
+                format!("{}", "Unsupported".red())
+            };
+
+            let msg = format!("{}: {}", self.name, msg);
+
+            OkTestResult { success, msg }
+        }
+    }
+
+    const SEV_CPUIDS: &[CpuId] = &[
+        CpuId {
+            name: "\nCPU Manufacturer",
```

Comment:
Let the caller decide when to introduce a `\n`. It shouldn't be included in this cpuid name.